tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -360,6 +360,13 @@ ROOT::Experimental::Internal::RPageSink::SealPage(const RPage &page, const RColu
    return SealPage(page, element, compressionSetting, fCompressor->GetZipBuffer());
 }
 
+void ROOT::Experimental::Internal::RPageSink::CommitDataset()
+{
+   for (auto cb : fOnDatasetCommitCallbacks)
```

Comment:
Maybe use `const auto &` or `auto &&` to avoid copying `std::function`?
Currently the copy is cheap, as the registered callback only saves a `this` pointer, but that might change in the future.